src/services/1inch.js

```diff
@@ -133,40 +131,11 @@ export const create1inchOrder = async (
   };
 };
 
-export const create1inchAllowanceTx = async (fromAssetAddress: string, clientAddress: string): Promise<Object> => {
-  if (!clientAddress) {
-    reportOrWarn('Unable to set allowance', null, 'error');
-    return null;
-  }
-
-  const abiFunction = [{
-    name: 'approve',
-    outputs: [{ type: 'bool', name: 'out' }],
-    inputs: [{ type: 'address', name: '_spender' }, { type: 'uint256', name: '_value' }],
-    constant: false,
-    payable: false,
-    type: 'function',
-    gas: 38769,
-  }];
-
-  const encodedContractFunction = encodeContractMethod(
-    abiFunction,
-    'approve',
-    [EXCHANGE_ADDRESS, ethers.constants.MaxUint256.toString()],
-  );
-
-  const txCount = await ethProvider().getTransactionCount(clientAddress);
-
-  return {
-    nonce: txCount.toString(),
-    to: fromAssetAddress,
-    gasLimit: '0',
-    gasPrice: '0',
-    chainId: '1',
-    value: '0',
-    data: encodedContractFunction,
+export const create1inchAllowanceTx =
+  async (fromAssetAddress: string, clientAddress: string): Promise<AllowanceTransaction | null> => {
+    const allowanceTx = await createAllowanceTx(fromAssetAddress, clientAddress, EXCHANGE_ADDRESS);
```

Comment:
Could this error? Just thinking if we can catch the error - even if we return the error, at least we can report then return the error.

Reply:
Potential errors are caught and reported inside `createAllowanceTx `